### ISSUE TITLE
Implement taking XBs and config generating for --test_mode

### ISSUE THREAD

**Issue by ShahriyarR** [ShahriyarR/MySQL-AutoXtraBackup#104]
This part of task will include some subtasks:

* The thing is we need 2 version of Xtrabackup:
2.3.x and 2.4.x debug build, can take them from Jenkins. -> DONE
* Extract them to `target` folder [if not exists already] -> DONE
* Generate config files -> DONE

It should reside in test directory.

> Note:
> For PS 5.7.x 2.4 version should be tested.
> For PS 5.6.x 2.4 and 2.3 both should be tested.

That means for each version the separate config file should be generated respectively.

**Comment by ShahriyarR** [comment]
```
$ python -m pytest -s -vvv ~/MySQL-AutoXtraBackup/test/test_prepare_env_test.py::TestBackupTestMode::test_get_xb_packages
========================================================================== test session starts ==========================================================================
platform linux -- Python 3.5.3, pytest-3.2.3, py-1.4.34, pluggy-0.4.0 -- /home/shahriyar.rzaev/virtualenvs/py_3_5_3_autoxtrabackup/bin/python
cachedir: .cache
rootdir: /home/shahriyar.rzaev/MySQL-AutoXtraBackup, inifile:
collected 1 item                                                                                                                                                         

test/test_prepare_env_test.py::TestBackupTestMode::test_get_xb_packages PASSED
```

**Comment by ShahriyarR** [comment]
```
$ python -m pytest -s -vvv ~/MySQL-AutoXtraBackup/test/test_prepare_env_test.py::TestBackupTestMode::test_extract_xb_archive
========================================================================== test session starts ==========================================================================
platform linux -- Python 3.5.3, pytest-3.2.3, py-1.4.34, pluggy-0.4.0 -- /home/shahriyar.rzaev/virtualenvs/py_3_5_3_autoxtrabackup/bin/python
cachedir: .cache
rootdir: /home/shahriyar.rzaev/MySQL-AutoXtraBackup, inifile:
collected 1 item                                                                                                                                                         

test/test_prepare_env_test.py::TestBackupTestMode::test_extract_xb_archive PASSED
```

**Comment by ShahriyarR** [comment]
```
$ python -m pytest -rws --tb=long -vvv ~/MySQL-AutoXtraBackup/test/test_config_generator.py::TestConfigGenerator::test_generate_config_files
========================================================================== test session starts ==========================================================================
platform linux -- Python 3.5.3, pytest-3.2.3, py-1.4.34, pluggy-0.4.0 -- /home/shahriyar.rzaev/virtualenvs/py_3_5_3_autoxtrabackup/bin/python
cachedir: .cache
rootdir: /home/shahriyar.rzaev/MySQL-AutoXtraBackup, inifile:
collected 1 item                                                                                                                                                         

test/test_config_generator.py::TestConfigGenerator::test_generate_config_files PASSED
```

**Comment by ShahriyarR** [comment]
The whole issue is implemented:

```
$ cat xb_2_4.conf 
[MySQL]
mysql = /home/shahriyar.rzaev/XB_TEST/server_dir/PS091017-percona-server-5.7.19-17-linux-x86_64-debug/bin/mysql
mycnf = 
mysqladmin = /home/shahriyar.rzaev/XB_TEST/server_dir/PS091017-percona-server-5.7.19-17-linux-x86_64-debug/bin/mysqladmin
mysql_user = root
mysql_password = 
#use either socket or port + host combination
mysql_socket = /home/shahriyar.rzaev/XB_TEST/server_dir/PS091017-percona-server-5.7.19-17-linux-x86_64-debug/socket.sock
#mysql_host = 127.0.0.1
#mysql_port = 3306
datadir = /home/shahriyar.rzaev/XB_TEST/server_dir/PS091017-percona-server-5.7.19-17-linux-x86_64-debug/data

[Backup]
#optional: set pid directory
pid_dir = /tmp/MySQL-AutoXtraBackup
tmp_dir = /home/shahriyar.rzaev/XB_TEST/mysql_datadirs
#optional: set warning if pid of backup us running for longer than x
pid_runtime_warning = 2 Hours
backupdir = /home/shahriyar.rzaev/XB_TEST/backup_dir/ps_5_7
backup_tool = /home/shahriyar.rzaev/XB_TEST/server_dir/target/percona-xtrabackup-2.4.x-debug/bin/xtrabackup
#optional: specify different path/version of xtrabackup here for prepare
#prepare_tool = 
xtra_prepare = --apply-log-only
#optional: pass additional options for backup stage
#xtra_backup = --compact
#optional: pass additional options for prepare stage
#xtra_prepare_options = --rebuild-indexes
#optional: pass general additional options; it will go to both for backup and prepare
#xtra_options = --binlog-info=ON --galera-info
#optional: set archive and rotation
#archive_dir = /home/shahriyar.rzaev/XB_TEST/backup_archives
#full_backup_interval = 1 day
#max_archive_size = 100GiB
#max_archive_duration = 4 Days
#optional: warning(enable this if you want to take partial backups). specify database names or table names.
#partial_list = test.t1 test.t2 dbtest

[Compress]
#optional
#enable only if you want to use compression.
#compress = quicklz
#compress_chunk_size = 65536
#compress_threads = 4
#decompress = TRUE
#enable if you want to remove .qp files after decompression.(not available yet, will be released with xb 2.3.7 and 2.4.6)
#remove_original = FALSE

[Encrypt]
#optional
#enable only if you want to create encrypted backups
#xbcrypt = /home/shahriyar.rzaev/XB_TEST/server_dir/target/percona-xtrabackup-2.4.x-debug/bin/xbcrypt
#encrypt = AES256
#please note that --encrypt-key and --encrypt-key-file are mutually exclusive
#encrypt_key = VVTBwgM4UhwkTTV98fhuj+D1zyWoA89K
#encrypt_key_file = /path/to/file/with_encrypt_key
#encrypt_threads = 4
#encrypt_chunk_size = 65536
#decrypt = AES256
#enable if you want to remove .qp files after decompression.(not available yet, will be released with xb 2.3.7 and 2.4.6)
#remove_original = FALSE

[Xbstream]
#experimental
#enable this, if you want to stream your backups
#xbstream = /home/shahriyar.rzaev/XB_TEST/server_dir/target/percona-xtrabackup-2.4.x-debug/bin/xbstream
#stream = xbstream
#xbstream_options = -x --parallel=100
#xbs_decrypt = 1
# warn, enable this, if you want to stream your backups to remote host
#remote_stream = ssh xxx.xxx.xxx.xxx

[Remote]
#optional remote syncing
remote_conn = root@xxx.xxx.xxx.xxx
remote_dir = /home/sh/Documents

[Commands]
start_mysql_command = /home/shahriyar.rzaev/XB_TEST/server_dir/PS091017-percona-server-5.7.19-17-linux-x86_64-debug/start
stop_mysql_command = /home/shahriyar.rzaev/XB_TEST/server_dir/PS091017-percona-server-5.7.19-17-linux-x86_64-debug/stop
chown_command = chown -R shahriyar.rzaev:shahriyar.rzaev

[TestConf]
gitcmd = --recursive --depth=1 https://github.com/percona/percona-server.git -b 5.7
testpath = /home/shahriyar.rzaev/XB_TEST/server_dir
```
